src/target.cr

```diff
@@ -9,12 +9,7 @@ module Shards
       main = nil
 
       pull.each_in_mapping do
-        case key = pull.read_scalar
-        when "main"
-          main = pull.read_scalar
-        else
-          # ignore unknown dependency mapping for future extensions
-        end
+        main = pull.read_scalar if pull.read_scalar == "main"
```

Comment:
This is a very questionable improvement.
In the original code the order of `read_scalar` calls is clearly aligned with the execution order. With this change it's confusing because the key is read first, but it's in a trailing condition.
So this should at least use a normal conditional, not a trailing one.

But I actually like the `case` statement. I guess it's not too important to keep it, but I don't see a real reason for changing it either.
*If* we change it, the comment should be kept, though.

Reply:
Ameba didn't like the unused `key` variable so I just switched it to something more compact, but I agree that the `case` statement gave it better readability. Perhaps a simple `if` statement instead? It feels like a bit of a waste to use `case` for one condition.

Reply:
Reverted in 4525063e95295a6081fc790529b8e41740168fb8.